frontend/src/metabase/admin/settings/components/widgets/FormattingWidget.tsx

```diff
@@ -0,0 +1,291 @@
+            <FormattingInput
+              id="currency"
+              label={t`Unit of currency`}
+              value={currency}
+              inputType="select"
+              options={currencyOptions.map(({ name, value }) => ({
+                label: name,
+                value,
+              }))}
+              onChange={(newValue) =>
+                handleChange({
+                  ...localValue,
+                  "type/Currency": {
+                    ...localValue?.["type/Currency"],
+                    currency: newValue as string,
+                  },
+                })
+              }
+            />
+            <FormattingInput
+              id="currency_style"
+              label={t`Currency label style`}
+              value={currencyStyle}
+              inputType="radio"
+              options={getCurrencyStyleOptions(currency)}
+              onChange={(newValue) =>
+                handleChange({
+                  ...localValue,
+                  "type/Currency": {
+                    ...localValue?.["type/Currency"],
+                    currency_style: newValue as string,
+                  },
+                })
+              }
+            />
+          </FormattingSection>
+        </Stack>
+      )}
+    </Stack>
+  );
+}
+
+function FormattingInput({
+  id,
+  label,
+  value,
+  onChange,
+  options,
+  inputType,
+}: {
+  id: string;
+  label: string;
+  value: any;
+  onChange: (newValue: string | boolean | number) => void;
+  options?: { label: string; value: string }[];
+  inputType: "boolean" | "select" | "radio";
+}) {
+  const [localValue, setLocalValue] = useState(value);
+
+  useEffect(() => {
+    setLocalValue(value);
+  }, [value]);
+
+  const handleChange = (newValue: string | boolean) => {
+    setLocalValue(newValue);
+    onChange(newValue);
+  };
+
+  return (
+    <Stack gap="sm" data-testid={`${id}-formatting-setting`}>
+      <Text htmlFor={id} component="label" fw="bold" display="block">
+        {label}
+      </Text>
+      {inputType === "select" && (
+        <Select
+          id={id}
+          value={localValue}
+          onChange={handleChange}
+          data={options ?? []}
+        />
+      )}
+      {inputType === "boolean" && (
+        <Switch
+          id={id}
+          checked={localValue}
+          onChange={(e) => handleChange(e.target.checked)}
+          label={localValue ? t`Enabled` : t`Disabled`}
+          w="auto"
+        />
+      )}
+      {inputType === "radio" && (
+        <Radio.Group id={id} value={localValue} onChange={handleChange}>
+          <Stack gap="sm">
+            {options?.map(({ label, value }) => (
+              <Radio key={value} value={value} label={label} />
+            ))}
+          </Stack>
+        </Radio.Group>
+      )}
+    </Stack>
+  );
+}
+
+function FormattingSection({
+  title,
+  children,
+}: {
+  title: string;
+  children: React.ReactNode;
+}) {
+  return (
+    <Stack gap="sm">
```

Comment:
could use a bit of margin I think

Old:
![Screenshot 2025-05-15 at 7 39 27 AM](https://github.com/user-attachments/assets/c659d4e5-19d9-4e23-a128-1900fb48fef7)

New:
![Screenshot 2025-05-15 at 7 39 22 AM](https://github.com/user-attachments/assets/4ba7b881-e334-4fb9-83c6-8ac225171ef8)

Reply:
Yeah that was a deliberate change because IMO the old one looked really gappy, I can change it back tho

Reply:
eh, the hierarchy on these pages is kind of a mess. let's not stress about it now. this will get much better in our next milestone.